### PR TITLE
Add TOBASCO controller

### PR DESCRIPTION
**Pull request by pckbls**
Hi,

here's the PR as discussed in issue #2 

Cheers
Patrick

### CODE REVIEW

File: controllers/TOBASCOController.py [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    def __repr__(self):
+        return '<TOBASCOController-%d>' % id(self)
+
+    def setPlayerFeedback(self, dict_params):
+        super(TOBASCOController, self).setPlayerFeedback(dict_params)
+
+        # append the data to our feedback history
+        self.feedback_hist.append(self.feedback)
+
+    def isBuffering(self):
```
Comment (ldecicco) [inline controllers/TOBASCOController.py:62]: That's the main reason to leave the implementation of isBuffering to the controller :), I wouldn't call it a hack, it's just the way you define your "buffering" phase.

File: controllers/TOBASCOController.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            #p_tilde_i = feedback["last_fragment_size"] / download_duration
+            p_tilde_i = feedback["bwe"]
+
+            sum_o += p_tilde_i * self.time_intersect(t_i_b, t_i_e, t1, t2)
+            sum_u += self.time_intersect(t_i_b, t_i_e, t1, t2)
+
+        return sum_o / sum_u
+
+    # the actual adaptation algorithm
+    def algo(self):
```
Comment (ldecicco) [inline controllers/TOBASCOController.py:180]: I would use a more explicative name here.

Comment (pckbls) [inline controllers/TOBASCOController.py]: For which variable? tau? Or for the entire function?

**Comment by ldecicco** [comment]
Hi Peter,

I've merged your pull request.

File: controllers/TOBASCOController.py (hunk 2 above)
Comment (ldecicco) [inline controllers/TOBASCOController.py:180]: I already did it (I was meaning the entire function name). The name of the function is now adaptationAlgorithm.

**Pull request merged**